verticapy/learn/tools.py

```diff
@@ -0,0 +1,611 @@
+#           \         /    /     /
+#            \_______/    /     /
+#             ______     /     /
+#             \    /    /     /
+#              \  /    /     /
+#               \/    /     /
+#                    /     /
+#                   /     /
+#                   \    /
+#                    \  /
+#                     \/
+#                    _
+# \  / _  __|_. _ _ |_)
+#  \/ (/_|  | |(_(_|| \/
+#                     /
+# VerticaPy is a Python library with scikit-like functionality to use to conduct
+# data science projects on data stored in Vertica, taking advantage Vertica’s
+# speed and built-in analytics and machine learning features. It supports the
+# entire data science life cycle, uses a ‘pipeline’ mechanism to sequentialize
+# data transformation operations, and offers beautiful graphical options.
+#
+# VerticaPy aims to solve all of these problems. The idea is simple: instead
+# of moving data around for processing, VerticaPy brings the logic to the data.
+#
+#
+# Modules
+#
+# VerticaPy Modules
+from verticapy.toolbox import *
+
+#
+# ---#
+def check_model(name: str, cursor=None):
+    """
+---------------------------------------------------------------------------
+Checks if the model already exists.
+
+Parameters
+----------
+name: str
+    Model name.
+cursor: DBcursor, optional
+    Vertica DB cursor.
+    """
+    check_types([("name", name, [str],)])
+    cursor, conn = check_cursor(cursor)[0:2]
+    cursor.execute("SELECT * FROM MODELS WHERE model_name='{}'".format(name))
+    result = cursor.fetchone()
+    if result:
+        if conn:
+            conn.close()
+        raise NameError("The model '{}' already exists !".format(name))
+    else:
+        try:
+            cursor.execute(
+                "SELECT * FROM verticapy.models WHERE LOWER(model_name) = '{}'".format(
+                    str_column(name)
+                )
+            )
+            result = cursor.fetchone()
+        except:
+            result = []
+        if result:
+            raise NameError("The model '{}' already exists !".format(name))
+    if conn:
+        conn.close()
+
+
+# ---#
+def load_model(name: str, cursor=None, test_relation: str = ""):
+    """
+---------------------------------------------------------------------------
+Loads a Vertica model and returns the associated object.
+
+Parameters
+----------
+name: str
+    Model Name.
+cursor: DBcursor, optional
+    Vertica DB cursor.
+test_relation: str, optional
+    Relation to use to do the testing. All the methods will use this relation 
+    for the scoring. If empty, the training relation will be used as testing.
+
+Returns
+-------
+model
+    The model.
+    """
+    check_types([("name", name, [str],), ("test_relation", test_relation, [str],)])
+    cursor = check_cursor(cursor)[0]
+    try:
+        check_model(name=name, cursor=cursor)
+        raise NameError("The model '{}' doesn't exist.".format(name))
+    except:
+        pass
```

Comment:
The code in the `load_model` function is very strange! To check that the model already exits in an expected situation, the `check_model` function is supposed throw an exception. It is very odd to use exceptions for the normal flow of a task. Exception are supposed to be used for stopping the execution of a task, not as a replacement of if-else statements.
It was way more straightforward if the the `check_model` function had returned a true or false value indicating the existence of a model. Then, a `raise` could be used properly for stopping the execution of the task.

Reply:
Thanks for the idea! It is done and it will be pushed after the v0.5.0 release.

Reply:
OK